internal/rbd/replicationcontrollerserver.go

```diff
@@ -517,6 +517,11 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 		}
 	}
 
+	err = checkHealthyPrimary(ctx, rbdVol)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
If there is an expected delay (as I understand from the commit message), should this not get retried?

Reply:
Yes, it will be retried by the caller for us it's an internal server error. even after promoting the system as not reached the state what we expected.